src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -0,0 +1,287 @@
+package org.broadinstitute.hellbender.utils.pairhmm;
+
+import org.apache.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.genotyper.ReadLikelihoods;
+import org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix;
+import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VectorLoglessPairHMM extends LoglessPairHMM {
+
+    protected final static Logger logger = Logger.getLogger(VectorLoglessPairHMM.class);
+    protected long threadLocalSetupTimeDiff = 0;
+    protected static long pairHMMSetupTime = 0;
+    protected final static Boolean runningOnMac = System.getProperty("os.name").toLowerCase().startsWith("mac");
+
+    //Used to copy references to byteArrays to JNI from reads
+    protected class JNIReadDataHolderClass {
```

Comment:
Why are these protected? I'm not very familiar with JNI code.  Is it necessary that these be more visible than private in order to make them available to JNI?

If this class is going to be final I think these should end up as either private or default visibility